secure_random/Makefile

```diff
@@ -0,0 +1,19 @@
+LIBSODIUM_VER_TAG=1.0.16
+
+TF_CFLAGS=$(shell python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_compile_flags()))' 2>/dev/null)
+TF_LFLAGS=$(shell python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))' 2>/dev/null)
+
+all: build-op
+
+build-lib-sodium:
+	# TODO CHECK IF SODIUM LIB EXISTS FIRST
+	git clone https://github.com/jedisct1/libsodium.git
+	cd libsodium && git checkout $(LIBSODIUM_VER_TAG)
+	cd libsodium && ./autogen.sh && ./configure --disable-shared --enable-static \
+		--disable-debug --disable-dependency-tracking --with-pic
+	$(MAKE) -C libsodium
+	$(MAKE) -C libsodium install
+	rm -Rf libsodium
+
+build-op:
```

Comment:
this should depend on the secure_random.cc file so it only builds when secure_random.cc changes. It could also depend on any file inside the `secure_random` file changing.

I'd also suggest we consider a different structure, something like `src/operations/secure_random` or similar so it's separated nicely from the rest of the project.

Reply:
How o we specify the platform we want to build for?

Reply:
> How do we specify the platform we want to build for?

Haven't worked on anything concretely for that yet but it will include building on the correct circle ci image and passed the `--plat-name` arg to the `python setup.py bdist_wheel` call like:

```
python setup.py bdist_wheel --plat-name macosx_10_7_x86_64 # for macos
```